### PR TITLE
Remove Helm 2 support

### PR DESCRIPTION
**Pull request by tomncooper**
### Type of change

Refactoring

### Description

Helm2 support ended in [November last year](https://helm.sh/blog/helm-2-becomes-unsupported/) so this PR removes the helm 2 build option.

### Checklist

- [X] Make sure all tests pass
- [X] Update documentation
- [X] Update CHANGELOG.md

### CODE REVIEW

File: Makefile
Comment (scholzj) [inline Makefile:15]: There is more stuff in this file ... check the targets `bridge_version`, `release_prepare`, `release_helm_version`, `helm_pkg` ... they all need to be fixed.

**Review by scholzj — COMMENTED** [review]
Right now, we expected two binaries for the build: `helm2` for Helm 2 and `helm3` for Helm 3. Should we update this and move to using only `helm` with Helm 3? It might be easier for users to use since they would be able to use the default Helm and not some manually created names.

**Comment by tomncooper** [comment]
Good point, fixing now.

**Review by scholzj — APPROVED** [review]
Looks good. Lets see if all the tests pass.

**Comment by scholzj** [comment]
```
Run 'helm init' to configure helm.
+ echo 'Verifying the installation of helm3 binary...'
+ helm3 --help
Verifying the installation of helm3 binary...
.azure/scripts/setup-helm.sh: line 18: helm3: command not found
+ helmCommandOutput=127
+ '[' 127 '!=' 0 ']'
+ echo 'helm binary hasn'\''t been installed properly - exiting...'
+ exit 1
helm binary hasn't been installed properly - exiting...
```

**Comment by tomncooper** [comment]
Sorry I am still fixing some issues with the main make file.

**Comment by scholzj** [comment]
Can you resolve the conflicts so that I can run tests on it?

**Comment by tomncooper** [comment]
Done

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
Everything passed ... so I guess a good job. I will merge it after we get all the needed approvals.

**Review by Frawless — APPROVED** [review]

File: Makefile
Comment (ppatierno) [inline Makefile:76]: as done by @scholzj in this PR https://github.com/strimzi/strimzi-kafka-operator/pull/4225, we need to escape the "dot" in quay.io `'/name: quay\.io\/kafka-bridge/{n;s/\(tag: \).*/\1$(BRIDGE_VERSION)/g}'`

File: install/Makefile
Comment (ppatierno) [inline install/Makefile:32]: at this point I think we could even renaming from `crd_install_helm3` to just `crd_install_helm`

**Review by ppatierno — CHANGES_REQUESTED** [review]

File: install/Makefile
Comment (tomncooper) [reply install/Makefile:32]: I did think about doing that but I guess at some point there will be a helm 4 and we may need to run both again. If we leave it like this there is less to change. WDYT?

File: Makefile
Comment (tomncooper) [reply Makefile:76]: Good catch! I have patched that.

File: install/Makefile
Comment (ppatierno) [reply install/Makefile:32]: Makes sense. I was wondering we were going to support just one Helm version from now on. If not, then it's fine helm3 ;-)

**Review by ppatierno — APPROVED** [review]

File: install/Makefile
Comment (scholzj) [reply install/Makefile:32]: Well, I guess there will be a long time until Helm4. So I do not think that is something what we will add in next months. (at the same time, I do not have problem to keep the target called helm3)

**Pull request merged**
